scope.go

```diff
@@ -810,6 +820,44 @@ func (c *cluster) getReplica() *replica {
 	return r
 }
 
+// getHostSticky returns host by stickiness from replica.
+//
+// Always returns non-nil.
+func (r *replica) getHostSticky(sessionId string) *host {
+	idx := atomic.AddUint32(&r.nextHostIdx, 1)
+	n := uint32(len(r.hosts))
+	if n == 1 {
+		return r.hosts[0]
+	}
+
+	idx %= n
+	h := r.hosts[idx]
+
+	// Scan all the hosts for the least loaded host.
+	for i := uint32(1); i < n; i++ {
+		tmpIdx := (idx + i) % n
+
+		// handling sticky session
+		if sessionId != "" {
```

Comment:
it is better to make this check in the beginning of the function. Otherwise, if sessionId=="" it makes no difference here and host from L:834 will be returned anyway.

Reply:
removed redundant check, under scope.go L:59 there was verification already which avoided getting into getHostSticky function.